src/component/contents/exploration/DraftEditorBlockNode.react.js

```diff
@@ -0,0 +1,370 @@
+  customStyleMap: Object,
+  decorator: ?DraftDecoratorType,
+  direction: BidiDirection,
+  editorKey: string,
+  editorState: EditorState,
+  forceSelection: boolean,
+  selection: SelectionState,
+  startIndent?: boolean,
+  tree: List<any>,
+};
+
+/**
+ * Return whether a block overlaps with either edge of the `SelectionState`.
+ */
+const isBlockOnSelectionEdge = (
+  selection: SelectionState,
+  key: string,
+): boolean => {
+  return selection.getAnchorKey() === key || selection.getFocusKey() === key;
+};
+
+/**
+ * We will use this helper to identify blocks that need to be wrapped but have siblings that
+ * also share the same wrapper element, this way we can do the wrapping once the last sibling
+ * is added.
+ */
+const shouldNotAddWrapperElement = (
+  block: BlockNodeRecord,
+  contentState: ContentState,
+): boolean => {
+  const nextSiblingKey = block.getNextSiblingKey();
+
+  return nextSiblingKey
+    ? contentState.getBlockForKey(nextSiblingKey).getType() === block.getType()
+    : false;
+};
+
+const applyWrapperElementToSiblings = (
+  wrapperTemplate: *,
+  Element: string,
+  nodes: Array<React.Node>,
+): Array<React.Node> => {
+  const wrappedSiblings = [];
+
+  // we check back until we find a sibbling that does not have same wrapper
+  for (const sibling: any of nodes.reverse()) {
+    if (sibling.type !== Element) {
+      break;
+    }
+    wrappedSiblings.push(sibling);
+  }
+
+  // we now should remove from acc the wrappedSiblings and add them back under same wrap
+  nodes.splice(nodes.indexOf(wrappedSiblings[0]), wrappedSiblings.length + 1);
+
+  const childrenIs = wrappedSiblings.reverse();
+
+  const key = childrenIs[0].key;
+
+  nodes.push(
+    React.cloneElement(
+      wrapperTemplate,
+      {
+        key: `${key}-wrap`,
+        'data-offset-key': DraftOffsetKey.encode(key, 0, 0),
+      },
+      childrenIs,
+    ),
+  );
+
+  return nodes;
+};
+
+const getDraftRenderConfig = (
+  block: BlockNodeRecord,
+  blockRenderMap: DraftBlockRenderMap,
+): * => {
+  const configForType =
+    blockRenderMap.get(block.getType()) || blockRenderMap.get('unstyled');
+
+  const wrapperTemplate = configForType.wrapper;
+  const Element =
+    configForType.element || blockRenderMap.get('unstyled').element;
+
+  return {
+    Element,
+    wrapperTemplate,
+  };
+};
+
+const getCustomRenderConfig = (
+  block: BlockNodeRecord,
+  blockRendererFn: BlockRenderFn,
+): * => {
+  const customRenderer = blockRendererFn(block);
+
+  if (!customRenderer) {
+    return {};
+  }
+
+  const {
+    component: CustomComponent,
+    props: customProps,
+    editable: customEditable,
+  } = customRenderer;
+
+  return {
+    CustomComponent,
+    customProps,
+    customEditable,
+  };
+};
+
+const getElementPropsConfig = (
+  block: BlockNodeRecord,
+  editorKey: string,
+  offsetKey: string,
+  blockStyleFn: BlockStyleFn,
+  customConfig: *,
+): Object => {
+  let elementProps: Object = {
```

Comment:
Again, I don't know what flow internals are around props, but using a CovariantObject is safer, so perhaps getElementPropsConfig should return the type {+[key: string]: any};

Reply:
Been trying to play safe on this PR since I don't want to get blocked by flow types, a lot of times flow internal is behaving different than here so I think we can create a follow up PR / issue to harden up draft.js typings in general